### PR TITLE
Add Unicode support to CLI on Windows

### LINKED ISSUE #8305: CLI: Unicode passwords handling is broken
## Overview
So, just found out, that the cli on windows does not escape password correctly! Don't know if there other characters, but at least 'ç' does not work! So its definitly a bug. 

## Steps to Reproduce
Try to create new database with the cli-version of keepassXC
When it promts me to type in password, everything works as acpected. 
But when I want to open afterwards the database over the GUI, it does not work. When I try to create a database with a password which does not contain the caracter 'ç', everything works!

### PR DESCRIPTION
**Pull request by mdonoughe**
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

On Windows APIs, text is stored in 16-bit wide strings or 8-bit ANSI strings. If you use byte-oriented APIs such as QFile to interact with stdin/stdout/stderr, by default Windows will try to encode text using the ANSI code page. For most users, that's 1252, which cannot represent the characters users typically think of as "Unicode." For backwards compatibility reasons, Windows does not globally switch the ANSI code page to UTF-8, even though UTF-8 has become the de facto 8-bit character encoding.

This PR [sets the process code page to UTF-8](https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page) for keepassxc-cli.exe (supported since Windows 10 1903) and then ensures that the code page used for console input and output matches the process code page (QTextStream uses the ANSI code page by default and will automatically pick up UTF-8). On older versions of Windows, the code should still work, but only for characters that can be represented by the system ANSI code page.

Fixes #8305 

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )

image

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )

I added an account to the, previously empty, NonAscii.kbdx file and added a new test that executes the CLI binary to read the password. Unlike the other CLI tests, this one needs to execute the CLI as a separate process because of the way the console interaction works.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

### CODE REVIEW

**Comment by droidmonkey** [comment]
Very nice!

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,7 @@ include_directories(
 add_definitions(-DQT_TEST_LIB)
 
 set(KEEPASSX_TEST_DATA_DIR ${CMAKE_CURRENT_SOURCE_DIR}/data)
+add_compile_definitions(KEEPASSX_CLI_PATH="$<TARGET_FILE:keepassxc-cli>")
```
Comment (droidmonkey) [inline tests/CMakeLists.txt]: This appears to be a newer cmake function, is there an older equivalent?

**Comment by mdonoughe** [comment]
The coverage run failed because of an error in testcli, but it was in testAdd, which I didn't change, on Linux, where my changes to the CLI itself should be removed by the preprocessor.

**Comment by droidmonkey** [comment]
Probably a random anomaly, I reran it

**Review by droidmonkey — APPROVED** [review]

**Pull request merged**